docs/architecture/adr/0021-console-logging.md

```diff
@@ -0,0 +1,114 @@
+tags: [server]
+---
+
+# 0021 - Logging with the Console
+
+<AdrTable frontMatter={frontMatter}></AdrTable>
+
+## Context and Problem Statement
+
+As the server platform has matured so have the various logging extensions to support additional use
+cases and customer requests. [Serilog][serilog] is in place via shared logic for all services,
+initiated at startup, and over time additional its "sinks" with their needed configuration have been
+added as dependencies leaving this core logging layer with a growing list of conditions on how and
+when to use certain types of structured logging. Maintenance needs have grown to keep sink
+dependencies up to date and more are desired to be added. Some of the presently-available sinks have
+very little use and / or better alternatives now exist.
+
+## Considered Options
+
+:::note
+
+Bitwarden uses [Datadog][dd] as its monitoring tool and desires to increase its usage by engineers
+across the board to improve what we deliver.
+
+:::
+
+- **Maintain current logging options** - Support what is available today for logging methods and
+  expect those running the platform to configure what they need outside of it for log collection.
+- **Extend the plaform to specifically support Datadog** - A Serilog sink [exists][ddsink] and the
+  platform can send logs directly to Datadog.
+- **Consolidate logging providers** - Announce deprecation and migration plans for sinks not aligned
+  with core needs and center on the console for logs.
+
+## Decision Outcome
+
+Chosen option: **Consolidate logging providers**.
+
+### Positive Consequences
+
+- Streamlined logging experience across components.
+- Console logging fits well into container and orchestration tools.
+- Elimination of several third-party dependencies and their maintenance, as well as global settings.
+- No new dependencies that are merely aligned with the Bitwarden-specific cloud and its service
+  providers.
+
+### Negative Consequences
+
+- A small number of users will need to migrate to console or similar ingestion of logs.
+- The Admin Portal log browsing function will leave (if configured) in favor of using whatever is
+  configured for log processing.
+
+### Plan
+
+Using standard support policies, release notes will include a mention that three Serilog sinks will
+be removed:
+
+- CosmosDb
+- Sentry
+- Syslog
+
+The remaining sinks -- core functionality of Serilog -- will continue to be supported:
+
+- Console
+- File
+
+While the Serilog [console sink][serilogconsole] is currently an implicit dependency with what's
+provided for ASP.NET Core, it will be explicitly referenced.
+
+Solutions exist for users to shift processing of logs for the removed sinks to console or file and
+retain their integration. Admin Portal users can similarly continue to use CosmosDb for log
+retention, but it is suggested that application monitoring that's available be used instead that
+should in essentially all cases be able to receive and process console logs.
```

Comment:
I inferred that this work would also remove the CosmosDb sink is that not correct?

Reply:
Yes -- line 61 here.